apps/web/lib/web/views/opportunities_view.ex

```diff
@@ -13,4 +16,40 @@ defmodule Web.OpportunitiesView do
       _ -> gettext("difficulty-unknown")
     end
   end
+
+  @doc """
+  Generate the HTML for the toggable difficulty selection
+  """
+  def level_selectors(conn) do
+    current_selection = selected_levels(conn)
+
+    [level_selector(conn, 1, current_selection),
+     level_selector(conn, 5, current_selection),
+     level_selector(conn, 9, current_selection)]
+  end
+
+  defp level_badge(_level, false), do: "badge-disabled"
+  defp level_badge(level, true), do: "badge-#{level}"
+
+  defp level_selector(%{query_params: params}, level, current_selection) do
+    [badge, level_params] =
+      if level in current_selection do
```

Comment:
Just a generate though:

Since this is a learning example, in general, should we prefer case statements with boolean outcome + individual guards over if/else, or consider if/else something that is fine to use as well?

Personally I tend to do true/false cases over if/else, so just a thought.

Reply:
This was discussed offline, the suggested solution:

```elixir
defp badge_and_level(level, current_selection) when level in current_selection, do:
  [level_badge(level, true), current_selection -- [level]]
defp badge_and_level(level, current_selection), do:
  [level_badge(level, false), [level|current_selection]]

defp level_selector(%{query_params: params}, level, current_selection) do
  [badge, level_params] = badge_and_level(level, current_selection)

  query_params = Map.put(params, "level", Enum.join(level_params, ","))

  :span
  |> content_tag(translated_difficulty(%{level: level}), class: "badge #{badge}")
  |> link(to: Web.Router.Helpers.opportunities_path(Endpoint, :index, query_params))
end
```

Doesn't compile so we're going to leave this as is for now

```shell
** (ArgumentError) invalid args for operator “in”, it expects a compile-time list or compile-time range on the right side when used in guard expressions, got: current_selection
```